### ISSUE TITLE
No White Spectrum

### ISSUE THREAD

**Issue by frankyboy1973** [datech/node-red-contrib-amazon-echo#11]
Since 0.1.6 no White Spectrum avalible and Color goes always back to Peppermint Creme

**Comment by datech** [comment]
Hello, 

Is there any debug messages on the Node-Red console?

**Comment by wongnam** [comment]
@frankyboy1973 Are you talking about Color Capabilities in Alexa? Including Color Temperature?
What kind of you Echo device? Echo, Echo Dot 1, Echo Dot 2, Echo Dot 3, Echo Plus.. this is important.
@datech If you have free time should create a template for every on to select or answer and provide enough information at one time so that we can easy to imaging the issue and no need to ask one by one again.

**Comment by frankyboy1973** [comment]
@datech No, there is no Debug Message in Node Red or in Ioroker Log.
@wongnam Sorry, i am from Germany and my Engish is not very good.
In the Alexa App under Version 0.1.4 it was possible to choose Color Temperature like Warm, Warmwhite, Cold, Coldwhite, now, in the new Version i can only chosse Colors. But when i give a speak Command to Alexa, to turn Light Warm or White or Cold it works, the CT Out changes.
When i choose a Color in the App or per Speak Command nothing happens, the Hue Output is alsways 0. In the App the Color goes always back to Pefferminz Creme. In Version 0.1.4 it was working, but not in the 0.1.6.
It was not possible to update from 0.1.4 to 0.1.6. Error, not installed. I deinstalled it and then I installed it new.
I have Echo Dot 2, Echo 2, Echo Plus, Echo Spot, Echo Show 2.

**Comment by datech** [comment]
@frankyboy1973 Versions 0.1.5 and 0.1.6 just enabled the support for older Echo devices.

May I ask you to set the Node-Red logging to debug and check the console again.
https://nodered.org/docs/user-guide/logging

Also, can you test the following:
1. Delete all devices using the Alexa app
2. Turn off all Echo devices except Echo 2
3. Start the discovery procedure again
4. Test the color capabilities

@wongnam Thanks. It is in my plan to create an issue template

**Comment by wongnam** [comment]
@datech the change since 0.1.5 that is using node-ssdp, i don't know whether the node-ssdp will effect to the Color Capabilities or not, special for Echo Dot 2nd gen. Please note that. Thanks.

**Comment by wongnam** [comment]
I've just delete my comment. Sorry to ignore my words.

**Comment by frankyboy1973** [comment]
Sorry, but i dont know how i can edit/open the settings.js, i am not a pro on linux. I have found it in the Node Red Folder, but dont know how to open it.
Yes i can delete the diveses and try it with only Echo 2 online.

**Comment by frankyboy1973** [comment]
Ok, when i discover the diveces only with Echo 2 online, the color Temperature is avalible in the App and the Colors capabilities is ok, the Hue Output changes, and the choosen Color is shown in the App. so i can say, the Funktion of the Node is OK. When i search with only Dot 2 online, the Color Temperature is not avalible in the App and the Color doesent change.
Perhaps you can help me by another Problem, i am using Ikea Trafri and i want to convert the Hue Output to Rgb or Hex, any Idear how i can do this. The Bri and Ct output is no Problem, i have already done that.

**Comment by datech** [comment]
@frankyboy1973 Are you using the latest software version for Echo Dot 2nd gen?   
https://www.amazon.com/gp/help/customer/display.html?nodeId=201602210

Would it be possible to update to latest version if you haven't already done so?

About hue & saturation conversion:
You can check the hslToRgb function in this example: https://gist.github.com/mjackson/5311256

**Comment by frankyboy1973** [comment]
Yes, i am using the latest Software Version for Echo Dot Gen.2 (628568520)

**Comment by frankyboy1973** [comment]
Now the Hue Output is always 43690, i tought the Hue must be between 0 and 360.

**Comment by wongnam** [comment]
Yep, I also enounter this situation dueto multiple Echo in a same network.

I don't know which Echo will discover which Node :( , so i am in passive for this case.

**Comment by datech** [comment]
Just released v0.1.7 with updated model of simulated light. It wasn't tested on Dot 2nd gen and I'm not sure that it will fix the support of color settings.

@frankyboy1973 
Saturation of the light (min 0, max 254)
Hue of the light (min 0, max 65535)

**Comment by wongnam** [comment]
@datech Echo Dot 2nd generation can found "color capabilities" in Aleaxa app but it is unusable due to it's not send value to the node output, it also no "color temperature" capabilities.
![image](https://user-images.githubusercontent.com/36041520/52164110-c9d21000-271e-11e9-9e44-b95e1b4956f1.png)

**Comment by datech** [comment]
@frankyboy1973 , @wongnam Have you tried with a new device added to the flow?

**Comment by frankyboy1973** [comment]
Hi, i deletet all devices in Alexa and all Nodes in Node Red and tried it with new Nodes.

**Comment by wongnam** [comment]
@datech I have just added a new device to the flow, the issue still persist.

**Comment by datech** [comment]
[v0.1.8](../releases/tag/v0.1.8) was released. Color capabilities has to be working now.

Could you try again with a new node added to the flow? 

I haven't done any fixes about color temperature capabilities, so I expect that they might be working with voice commands and still can be missing from the app. More important to me is how the current release is working with multiple types and generations of Echo devices.

**Comment by datech** [comment]
Thanks for checking this out.

I think that color is returned back to peppermint, because Echo Dot 2nd gen is ignoring or doesn't understand sat/hue color settings. I will add "xy" color in the API response in the next version, so it might fix this issue.

**Comment by frankyboy1973** [comment]
Another problem for me, is that the brightness of the devices nodes goes to 100%, after Node Red restart. When i say Alexa to turn light on, all Lights are on full brightness.

**Comment by frankyboy1973** [comment]
In the settings.js the context is configured to localfilesystem. Is it perhaps a problem that i run Node Red as an Iobroker Instance? Or i am at the wrong seetings.js? I have found it in /opt/iobroker/node_modules/node-red. Sorry, i am not a pro in these things.
